packages/nimble-components/src/table/index.ts

```diff
@@ -1,6 +1,17 @@
+import { observable } from '@microsoft/fast-element';
 import { DesignSystem, FoundationElement } from '@microsoft/fast-foundation';
+import {
+    ColumnDef,
+    TableState,
+    Updater,
+    Table as TanstackTable,
```

Comment:
Another alternative would be to create a separate class whose responsibility is to interface with TanStack and encapsulate its API so that the nimble-table index.ts doesn't contain any TanStack types. Whether this is worthwhile probably depends on how big we expect this class to get and how much of it we expect to be TanStack-related. I don't have a sense for this so I'll let you decide whether to do this now, later, or never.

Reply:
After considering the option to move TanStack-specific code into its own class (and discussing it with @atmgrifter00), I've decided not to do that at this time. It's not immediately clear where the separation of logic would/should be, and I think it'll add additional overhead in our development. I'm not against doing it at some point in the future if we feel it will provide value.

For now, I'll rename the imported types to better distinguish TanStack types from types we are creating.